### ISSUE TITLE
Type error when extracting computedFields out from the defineDocumentType function as a const

### ISSUE THREAD

**Issue by Itsnotaka** [contentlayerdev/contentlayer#348]
```ts
/** @type {import('contentlayer/source-files').ComputedFields} */
const computedFields = {
  slug: {
    type: "string",
    resolve: (doc: { _raw: { flattenedPath: never } }) =>
      `/${doc._raw.flattenedPath}`,
  },
  slugAsParams: {
    type: "string",
    resolve: (doc: { _raw: { flattenedPath: string } }) =>
      doc._raw.flattenedPath.split("/").slice(1).join("/"),
  },
};

export const Posts = defineDocumentType(() => ({
  name: "Posts",
  filePathPattern: `posts/**/*.mdx`,
  contentType: "mdx",
  fields: {
    title: {
      type: "string",
      required: true,
    },
    description: {
      type: "string",
    },
    date: {
      type: "date",
      required: true,
    },
    published: {
      type: "boolean",
      default: true,
    },
    image: {
      type: "string",
      required: true,
    },
    visibility: {
      type: "string",
      required: true,
    },
    authors: {
      // Reference types are not embedded.
      // Until this is fixed, we can use a simple list.
      // type: "reference",
      // of: Author,
      type: "list",
      of: { type: "string" },
      required: true,
    },
  },
  // @ts-expect-error TODO: it's conflict with extracted computedFields (This would introduce the error)
  computedFields,
}));
```

By using the code of the above in `contentlayer.config.ts` it gives type error. However putting the constant object back fixes this.

**Comment by nkilm** [comment]
#### @Itsnotaka Please can you elaborate more about the error you are facing? i.e, which line, how to reproduce, file-formats of data etc.

Also if you're using `.mdx` files then should'nt the logic for extracting `slug` be different?
```javascript
// this'll work
slug: {
      type: "string",
      resolve: (post) =>
        post._raw.sourceFileName
          // hello-world.mdx => hello-world
          .replace(/\.mdx$/, ""),
    },
```

**Comment by schickling** [comment]
I understand the friction you're facing and agree that it's not ideal. Would having a `defineComputedFields` function help for your situation?

**Comment by Itsnotaka** [comment]
yeah it would be helpful.

**Comment by schickling** [comment]
This should be addressed with the [`0.3.0` release](https://github.com/contentlayerdev/contentlayer/releases/tag/v0.3.0). 🎉
